Shared/Samples/Create buffers around points/CreateBuffersAroundPointsView.swift

```diff
@@ -57,29 +68,30 @@ struct CreateBuffersAroundPointsView: View {
                     .disabled(model.bufferPoints.isEmpty)
                 }
             }
-            .alert("Buffer Radius", isPresented: $model.inputBoxIsPresented, actions: {
-                TextField("100", text: $model.radiusInput)
+            .alert("Buffer Radius", isPresented: $inputBoxIsPresented, actions: {
+                TextField("100", text: $radiusInput)
                     .keyboardType(.numberPad)
                 Button("Done") {
-                    // Check to ensure the tapPoint is within the boundary.
-                    guard model.boundaryContains(model.tapPoint) else {
+                    defer { radiusInput.removeAll() }
+                    
+                    // Ensure the tapPoint is within the boundary.
+                    guard let tapPoint, model.boundaryContains(tapPoint) else {
                         status = .outOfBoundsTap
                         return
                     }
                     
                     // Ensure that the input is valid.
-                    if let radius = Double(model.radiusInput),
-                       radius > 0 && radius < 300 {
-                        model.addBufferPoint()
-                        model.drawBuffers()
-                        status = .bufferCreated
-                    } else {
+                    guard let radius = Double(radiusInput),
+                          radius > 0 && radius < 300 else {
                         status = .invalidInput
+                        return
                     }
-                    model.radiusInput = ""
+                    model.addBuffer(point: tapPoint, radius: radius)
+                    model.drawBuffers(shouldUnion: shouldUnion)
+                    status = .bufferCreated
```

Comment:
One other thing. The else clause of guard statements wasn't intended for doing additional work. In my experience, doing so _always_ harms readability. It breaks the mental model of "forward progress is halted". Here, an initial guard statement followed by an if statement would make this closure much more readable:

```swift
guard let tapPoint else {
    preconditionFailure("Missing tap point")
}

let newStatus: Status
if model.boundaryContains(tapPoint) {
    if let radius = Double(radiusInput),
       radius > 0 && radius < 300 {
        model.addBuffer(point: tapPoint, radius: radius)
        model.drawBuffers(shouldUnion: shouldUnion)
        newStatus = .bufferCreated
    } else {
        newStatus = .invalidInput
    }
} else {
    newStatus = .outOfBoundsTap
}
status = newStatus
```

Reply:
I agree, this is a good point and I've had put it in the style guide before.

Would you say that going on we should limit the `else` branch to only 1 line, that can be one of the following

- return/return nil/return sth empty or default
- throw
- fatalError
-

Reply:
A guard's else clause can also continue a loop. But that is all part of the documentation for guard statements, inherent to how they function. I would just say that a guard's else clause shouldn't perform any work other than what is needed. Its job is to get out of the current scope and it should do so as quickly as possible. By "other work", I mean work with consequences. There are legitimate use cases for having additional code in a guard's else clause, such as logging. But even that, in my experience, is usually just temporary for debugging.